core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -653,7 +655,10 @@ void addEdge(int fromIndex, int toIndex, PointList pointList, IntsRef flags, Rea
         if (doSimplify && pointList.size() > 2)
             simplifyAlgo.simplify(pointList);
 
-        double towerNodeDistance = distCalc.calcDistance(pointList);
+        // todonow: how should we manage geometry encoded values / where do they come from, probably EncodingManager somehow?
+        //          here they are very similar to tag parsers. (handleWayTags -> handleGeometry)
+        GeometryEncodedValue distanceEnc = Distance.create();
+        double towerNodeDistance = distanceEnc.calculateDecimal(false, flags, pointList);
```

Comment:
If we ever replaced distance with the distance EV this would be the place to calculate the distance (and same for elevation gain etc.) in OSMReader.